src/routers/alpha-router/functions/best-swap-route.ts

```diff
@@ -465,12 +466,18 @@ export async function getBestSwapRouteBy(
     if (v2GasModel == undefined && v3GasModel == undefined) {
       throw new Error("Can't compute L1 gas fees.");
     } else {
+      // Now after v2 deploy everywhere, it's possible to to have v2 and v3 routes on L2 swap.
+      // So that we need to add both gas cost one by one
       const v2Routes = bestSwap.filter((routes) => routes.protocol === Protocol.V2);
       if (v2Routes.length > 0 && V2_SUPPORTED.includes(chainId)) {
         if (v2GasModel) {
           const v2GasCostL1 = await v2GasModel.calculateL1GasFees!(v2Routes as V2RouteWithValidQuote[]);
           gasCostsL1ToL2.gasUsedL1 = gasCostsL1ToL2.gasUsedL1.add(v2GasCostL1.gasUsedL1);
           gasCostsL1ToL2.gasUsedL1OnL2 = gasCostsL1ToL2.gasUsedL1OnL2.add(v2GasCostL1.gasUsedL1OnL2);
```

Comment:
This naming is very hard to follow and I know that some of it was already like this but would rly appreciate variable names that are more intuitive. Maybe destructing the return value from `v2GasModel.calculateL1GasFees` and instead of doing `.add` you can probably just set `gasCostsL1ToL2.gasUsedL1  = gasUsedL1`?

Reply:
it seems like there is a lot more logic happening here than there actually is, making it harder to review

Reply:
added more explanations https://github.com/Uniswap/smart-order-router/pull/485/files#diff-bf0f39dc98ca22700a5c33efc44d6acf65a840bcface7d39297ffa99d20e9478R479-R483

tbh, meanwhile I can see the path to full readability (by refactoring v2 and v3 gas models for the L2toL1 security fee part), I don't think the time permits for that amount of refactoring.